test/linter/test-versions.js

```diff
@@ -89,20 +89,13 @@ function checkVersions(supportData, relPath, logger) {
             typeof statement.version_removed === 'string'
           ) {
             if (
-              (statement.version_added.startsWith('≤') &&
-                statement.version_removed.startsWith('≤') &&
-                compareVersions.compare(
-                  statement.version_added.replace('≤', ''),
-                  statement.version_removed.replace('≤', ''),
-                  '<',
-                )) ||
-              ((!statement.version_added.startsWith('≤') ||
-                !statement.version_removed.startsWith('≤')) &&
-                compareVersions.compare(
-                  statement.version_added.replace('≤', ''),
-                  statement.version_removed.replace('≤', ''),
-                  '>=',
-                ))
+              compareVersions.compare(
+                statement.version_added.startsWith('≤')
+                  ? '0' // 0 was chosen as it's a number lower than any possible browser version
+                  : statement.version_added,
+                statement.version_removed.replace('≤', ''),
+                '>=',
+              )
```

Comment:
Apologies for the delay in my response, I've been away from BCD for a little bit due my contract expiring and catching up on some business maintenance stuff.  I've separated the logic into a separate function now, to help with legibility!  (I didn't want to do it before to avoid creating more functions than needed, but you're right, it certainly makes the main function's logic more complex than it needs to be.)